src/site/_includes/js/eligibility.js

```diff
@@ -133,6 +133,20 @@ const cnst = {
     // "Personal Property - GA Policy [180]"
     MAX_RESOURCES: 500, // USD
   },
+  homelessPreventionSystem: {
+    // https://preventhomelessness.org/#eligibility
+    ANNUAL_INCOME_LIMITS: [ // USD per year
+      102300,
+      116900,
+      131500,
+      146100,
+      157800,
+      169500,
+      181200,
+      192900,
+    ],
+    // Aditional person income limit not specified
```

Comment:
We should be sure to support > 8 people in a household.

The income limits given note that they are from the Department of Housing and Urban Development (HUD).  You can look up the income limits for Santa Clara county in 2024 for "low income" [here](https://www.huduser.gov/portal/datasets/il/il2024/2024summary.odn?states=%24states%24&data=2024&inputname=METRO41940M41940*0608599999%2BSanta+Clara+County&stname=California&statefp=06&year=2024&selection_type=county) and see they match.  

Unfortunately, the calculation for >8 people households is a bit complex.  If you click the "Click for More Detail" button at the link above for the low income category, you can get to this [page](https://www.huduser.gov/portal/datasets/il/il2024/2024ILCalc3080.odn?inputname=Santa+Clara+County&area_id=METRO41940M41940&fips=0608599999&type=county&year=2024&yy=24&stname=California&stusps=CA&statefp=06&ACS_Survey=%24ACS_Survey%24&State_Count=%24State_Count%24&areaname=%24passname%24&incpath=%24incpath%24&level=80) and see how the calculation is done for >8 people. (see number 5 on that page)

Luckily, I've already [coded this](https://github.com/theunitedeffort/theunitedeffort.org/blob/edaca0e2879c7f4690dd39f7bcfd61f3608c421f/src/site/_includes/js/eligibility.js#L2296-L2310) for the Housing Choice Voucher program.  The `MonthlyIncomeLimits` object can take a function for the `addlPersonExtra` parameter, and it'll use that to compute the limit instead of adding on a constant value.

The work now is going to be to refactor some of this code so that we can more easily re-use the relevant constants and calculation.  You might pull `extraCalc` out of `housingChoiceResult` (and probably give it a more useful name) and create a `common` or `base` property of the `cnst` object that can hold constants shared between programs (such as [these](https://github.com/theunitedeffort/theunitedeffort.org/blob/edaca0e2879c7f4690dd39f7bcfd61f3608c421f/src/site/_includes/js/eligibility.js#L139-L144))

You can do that refactor as a separate PR or keep it in this one; up to you.

Reply:
Are the requirements the same for both the Homelessness Prevention System and the Housing Choice Voucher? Such that I can copy the parameters over from `const cnst`?

Reply:
The requirements are not the same, but the calculation for how to get an income limit for households > 8 is the same between the two programs because the two programs both use HUD income limits.

You should be able to move those [four constants](https://github.com/theunitedeffort/theunitedeffort.org/blob/edaca0e2879c7f4690dd39f7bcfd61f3608c421f/src/site/_includes/js/eligibility.js#L139-L144) I mentioned out of `cnst.housingChoice` and into a new property called something like `cnst.common` or `cnst.base` and then use them in both `housingChoiceResult` and `homelessPreventionSystemResult`

Reply:
I've tried copying those four constants into `cnst.homelessPreventionSystem` and copying the `extraCalc` codeblock into `homelessPreventionResult` adjusting the names accordingly, but checks are all failing due to a logic input value error. I suspect this has to do with the `underIncomeLimit` constant but I'm not sure why the behavior should be different since I copied the code directly from `housingChoiceResult`.

Reply:
Ah, ok yes this was a bit complicated to figure out.  I had to remember how the tests are set up, and I admit it's confusing.

Here's an important thing to remember: All the program result functions (e.g. `homelessPreventionSystemResult`) simply take the `input` object and compute eligibility from only that.  The `input` object is just a javascript object.  In the production code, the `input` object gets _built_ from the html document and form input elements (the DOM).  _However_, in the eligibility-programs test, we hand-craft the default `input` object.  

So, eligibility-programs.test.js only tests the eligibility logic _given_ a certain `input` obj.  Whether or not the `input` object gets built correctly from the html form input elements is tested in eligibility-integration.test.js.  So, while you seem to be generating `input.unhousedRisk` correctly in `buildInputObj` at the end of the day it does not matter for eligibility-programs.test.js because that test never even calls `buildInputObj`.  Instead, you need to add it directly to the default `input` object defined for each test.  Put simply, you need to add

```
unhousedRisk: null,
```

in this [spot](https://github.com/theunitedeffort/theunitedeffort.org/blob/ba87b7db4370611da661a4a464b7e1c8920d8f74/src/site/_includes/js/test/eligibility-programs.test.js#L528).

The error being thrown due to an undefined input was because of `input.unhousedRisk` being undefined [here](https://github.com/theunitedeffort/theunitedeffort.org/blob/ba87b7db4370611da661a4a464b7e1c8920d8f74/src/site/_includes/js/eligibility.js#L2743).  I figured that out because the stacktrace of the test failure pointed me to [line 1682](https://github.com/theunitedeffort/theunitedeffort.org/blob/ba87b7db4370611da661a4a464b7e1c8920d8f74/src/site/_includes/js/eligibility.js#L1682), which is where all the eligibility conditions of a program get `and`ed together.  Since the error was not thrown during the calculation of `underIncomeLimit` I knew to focus my attention on `input.unhousedRisk`

Reply:
The `eligibility-integration.js` test below is failing as well:
```
    document.getElementById('unhoused-risk').checked = expected.unhousedRisk;
    expect(getInput().unhousedRisk).toBe(expected.unhousedRisk);
```
This is despite the fact I set `unhousedRisk` to `true` in `const expected`. `unhousedRisk` is also declared in `const inputData` in `eligibility.js`:
```
    unhousedRisk: getValueOrNull('unhoused-risk'),
```  
(line 2806)

Reply:
It's hard for me to debug without seeing the code in question, but one thing that comes to mind is that you need to be sure the integration test actually does the thing to make `unhousedRisk` equal to the expected value of `true`.  The test needs to do the equivalent of answering the "risk of losing housing" question "yes".  If the integration test does not answer the question (so to speak) then the value of `unhousedRisk` will correctly be `null`.

There is an [example](https://github.com/theunitedeffort/theunitedeffort.org/blob/edaca0e2879c7f4690dd39f7bcfd61f3608c421f/src/site/_includes/js/test/eligibility-integration.test.js#L1766-L1767) from a similar "pays utilities" question.